src/dashboard/apps/core/helpers.py

```diff
@@ -0,0 +1,58 @@
+"""Dashboard core helpers."""
+
+from typing import Optional
+
+from apps.auth.models import DashboardUser
+from apps.core.annuaire_entreprise_api import (
+    AnnuaireEntrepriseAPI,
+    CompanyAddressAdapter,
+    CompanyInformationAdapter,
+)
+from apps.core.models import Entity
+from apps.core.validators import validate_siret
+
+
+def sync_entity_from_siret(siret: str, user: Optional[DashboardUser] = None) -> Entity:
+    """Retrieve, Update or create and populate entity.
+
+    Retrieve and populate entity data based on SIRET input.
+    The function uses an external API to fetch company information and address using
+    the provided SIRET number.
+    If the entity does not already exist in the database, it will be created.
+    Optionally associates a user with the entity.
+
+    Arguments:
+        siret (str): The SIRET number used to identify and fetch the company's
+        information.
+        user (Optional[User]): A user instance to be associated with the created or
+        updated entity. Defaults to None.
```

Comment:
May it be `None` in a real-case scenario?

Reply:
I'm not sure I understand what you mean